assembly/src/library/masl.rs

```diff
@@ -69,9 +72,34 @@ impl MaslLibrary {
         Ok(Self {
             namespace,
             version,
+            locations,
             modules,
         })
     }
+
+    // STATE MUTATORS
+    // --------------------------------------------------------------------------------------------
+
+    /// Clears the source locations from this bundle.
+    pub fn clear_locations(&mut self) {
+        self.modules.iter_mut().for_each(|m| m.clear_locations())
+    }
+
+    // SERIALIZATION / DESERIALIZATION
+    // --------------------------------------------------------------------------------------------
+
+    /// Loads the [SourceLocation] of the procedures via [Module::load_locations].
+    pub fn load_locations<R: ByteReader>(
+        &mut self,
+        source: &mut R,
+    ) -> Result<(), DeserializationError> {
+        self.modules.iter_mut().try_for_each(|m| m.load_locations(source))
+    }
+
+    /// Writes the [SourceLocation] of the procedures via [Module::write_locations].
+    pub fn write_locations<W: ByteWriter>(&self, target: &mut W) {
+        self.modules.iter().for_each(|m| m.write_locations(target))
+    }
```

Comment:
Should we make these private? It feels like loading/writing locations should be fully encapsulated within the `MaslLibrary` struct.